### PR TITLE
Test TCP connection before upgrading database schema

### PR DESCRIPTION
**Pull request by ywk253100**
This commit moves the database schema upgrading after database initialization. The init will test TCP connection.

Signed-off-by: Wenkai Yin <yinw@vmware.com>

### CODE REVIEW

File: src/common/utils/utils.go [hunk 1]
```diff
@@ -88,15 +88,17 @@ func TestTCPConn(addr string, timeout, interval int) error {
...
 			select {
 			case <-cancel:
 				break
 			default:
-				conn, err := net.DialTimeout("tcp", addr, time.Duration(timeout)*time.Second)
+				conn, err := net.DialTimeout("tcp", addr, time.Duration(n)*time.Second)
 				if err != nil {
 					log.Errorf("failed to connect to tcp://%s, retry after %d seconds :%v",
 						addr, interval, err)
+					n = n * 2
```
Comment (reasonerjt) [inline src/common/utils/utils.go]: Do you intend to update the interval or the timeout?

**Comment by reasonerjt** [comment]
Please also resolve the conflict.

File: src/common/utils/utils.go (hunk 1 above)
Comment (ywk253100) [reply src/common/utils/utils.go]: It's used to update the timeout for every try, not the total timeout or interval.

**Comment by ywk253100** [comment]
@reasonerjt Updated.

**Pull request merged**